dataclass_function.py

```diff
@@ -92,3 +94,108 @@ class School(AsFrame):
     rrcp_school: Data[str]
     rrcp_area: Data[int]
     rrcp_country: Data[int]
+
+
+# function for getting list from df columns
+
+
+def createdf(file):
+    # read the file
+    df = pd.read_excel(file)
+
+    def clmnlist(i):
+        return list(df.iloc[:, i])
+
+    """ For the school df, showing:
+        school_id
+        rrcp_school
+        rrcp_area
+        rrcp_country
+    """
+    school_df = School.new(clmnlist(6), clmnlist(5), clmnlist(4), clmnlist(3))
+
+    """ For the teacher df, showing:
+        rred_user_id
+        red_rr_title
+        school_id
+    """
+    teach_df = Teacher.new(clmnlist(1), clmnlist(2), clmnlist(6))
+    """
+    I feel like below could be a for loop but not quite sure how
+    """
+    pupil_df = Pupil.new(

```

Comment:
then once you got to here, I expect you could do something like passing in all of the rows to the Pupil constructor which harry should be able to help with if it gets really hairy